src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -306,18 +306,34 @@ private TestClassInfo CreateClassInfo(Type classType, TestMethod testMethod)
 
         foreach (MethodInfo methodInfo in classType.GetTypeInfo().DeclaredMethods)
         {
-            // Update test initialize/cleanup method
-            UpdateInfoIfTestInitializeOrCleanupMethod(classInfo, methodInfo, false, instanceMethods);
+            if (KnownNonTestMethods.Methods.Contains(methodInfo.Name))
+            {
+                continue;
+            }
 
-            // Update class initialize/cleanup method
-            UpdateInfoIfClassInitializeOrCleanupMethod(classInfo, methodInfo, false, ref initAndCleanupMethods);
+            if (methodInfo.IsPublic && !methodInfo.IsStatic)
```

Comment:
This is done in the other place as well, but seems inconsistent with the rest of the code, where we inspect even shapes of methods that are obviously incorrect, and throw when they have incorrect shape.

So remove in both places?